src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -270,108 +241,47 @@ if (!product) {
-const personData = await personResponse.json();
-const randomPerson = personData.results[0];
----
-<p>{randomPerson.name.first}</p>
-```
+The HTTP method used in the request is available as `Astro.request.method`. This works like the browser's [`Request.method`](https://developer.mozilla.org/en-US/docs/Web/API/Request/method). It returns the string representation of the HTTP method used in the request.
 
-```astro title="src/components/RandomFact.astro"
+```astro title="src/pages/index.astro"
 ---
-const factResponse = await fetch('https://catfact.ninja/fact');
-const factData = await factResponse.json();
+console.log(Astro.request.method) // GET (when navigated to in the browser)
 ---
-<p>{factData.fact}</p>
 ```
 
+See more details about [`Astro.request`](/en/reference/api-reference/#astrorequest) in the API reference.
 
-The Astro page below using these components can render parts of the page sooner. The `<head>`, `<body>`, and `<h1>` tags are no longer blocked by data fetches. The server will then fetch data for `RandomName` and `RandomFact` in parallel and stream the resulting HTML to the browser.
+### Server Endpoints
 
-```astro title="src/pages/index.astro"
----
-import RandomName from '../components/RandomName.astro'
-import RandomFact from '../components/RandomFact.astro'
----
-<html>
-  <head>
-    <title>A name and a fact</title>
-  </head>
-  <body>
-    <h2>A name</h2>
-    <RandomName />
-    <h2>A fact</h2>
-    <RandomFact />
-  </body>
-</html>
-```
+A server endpoint, also known as an **API route**, is a special function exported from a `.js` or `.ts` file within the `src/pages/` folder. A powerful feature of server-side rendering on demand, API routes are able to securely execute code on the server.
 
-#### Including Promises directly
+The function takes an [endpoint context](/en/reference/api-reference/#endpoint-context) and returns a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response). 
 
-You can also include promises directly in the template. Instead of blocking the entire component, it will resolve the promise in parallel and only block the markup that comes after it.
+To learn more, see our [Endpoints Guide](/en/core-concepts/endpoints/#server-endpoints-api-routes).
 
-```astro title="src/pages/index.astro"
----
-const personPromise = fetch('https://randomuser.me/api/')
-  .then(response => response.json())
-  .then(arr => arr[0].name.first);
-const factPromise = fetch('https://catfact.ninja/fact')
-  .then(response => response.json())
-  .then(factData => factData.fact);
----
-<html>
-  <head>
-    <title>A name and a fact</title>
-  </head>
-  <body>
-    <h2>A name</h2>
-    <p>{personPromise}</p>
-    <h2>A fact</h2>
-    <p>{factPromise}</p>
-  </body>
-</html>
-```
 
-In this example, `A name` will render while `personPromise` and `factPromise` are loading.
-Once `personPromise` has resolved, `A fact` will appear and `factPromise` will render when it's finished loading.
+{/* 
+
+ */}
```

Comment:
I think we can delete this empty comment?

```suggestion
```

Reply:
Another Sarah Dark Pattern 👻